index.js

```diff
@@ -356,6 +356,19 @@
     ($.Undefined  = type0('Undefined'))
   ];
 
+  // Enum :: [Any] -> Type
+  $.Enum = function(constants) {
+    return {
+      type: 'sanctuary-def/Enum',
+      test: function(x) {
+        return $.Array(Any).test(constants) && constants.indexOf(x) !== -1;
```

Comment:
We should assume `constants` to be an Array. To support old versions of IE, though, we should avoid `Array.prototype.indexOf`. Also, it would be nice to use value-based equality. I suggest something like this:

```javascript
test: function(x) {
  var repr = show(x);
  var reprs = map(constants, show);
  for (var idx = 0; idx < reprs.length; idx += 1) {
    if (reprs[idx] === repr) return true;
  }
  return false;
}
```

Since `map(constants, show)` would then appear in both `test` and `toString`, we could even define `reprs` in the (innermost) scope shared by both functions.